Consider syncing with main branch of openzim/youtube

deldesir opened the pull request:
Hi George,

Your tubetools branch is currently not synced with the main branch of its upstream. I have made some changes to the codebase that I would like to propose for inclusion in it without diverging too much from openzim philosophy, but I am unable to do so unless the repo is synced with the upstream first.

Could you please consider syncing with the main branch of the upstream so that I can create a pull request with my proposed changes? I believe these changes will improve the functionality and performance of the project, and I would be grateful for the opportunity to contribute regularly.

Thank you for your consideration, and I look forward to discussing my changes with you further.

Sincerely,
Blondel

holta commented:
Thanks @deldesir and @georgejhunt !

QUESTION: Is it perhaps time for me to create repo https://github.com/iiab/youtube ?  To bring more visibility to this important work, and better align with upstream ?

georgejhunt commented:
sure

On Wed, Dec 21, 2022 at 10:54 AM A Holt ***@***.***> wrote:

> Thanks @deldesir <https://github.com/deldesir> and @georgejhunt
> <https://github.com/georgejhunt> !
>
> QUESTION: Is it perhaps time for me to create repo
> https://github.com/iiab/youtube ? To bring more visibility to this
> important work, and better align with upstream ?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/georgejhunt/youtube/pull/5#issuecomment-1361895736>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAOTQHC7TNGZUOWI6BHDW3LWONG6TANCNFSM6AAAAAATF5QCUE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

holta commented:
> sure

Done.  These 2 should be ready to go:

- https://github.com/iiab/youtube
- https://github.com/iiab/youtube/wiki

@deldesir & @georgejhunt LMK anything else that's needed :heavy_check_mark:

dan-niles committed:
Fix environment name for zimui dist

dan-niles committed:
Update README.md with unstable branch disclaimer and remove README.md in zimui folder

benoit74 committed:
Merge pull request #207 from openzim/new_ui_init

Move scraper to subfolder and initialize new Vue.js project

dan-niles committed:
Update scraper to generate JSON files for `zimui`

dan-niles committed:
Remove old jinja template files and  method in scraper.py

dan-niles committed:
Remove `locale` folder and files used for translation

benoit74 committed:
Merge pull request #219 from openzim/generate_json

Generate JSON files to be consumed by Vue.js UI

dan-niles committed:
Remove old font files and install through fontsource

dan-niles committed:
Add Pinia store for main module and API calls

dan-niles committed:
Setup routing and create channel home page

dan-niles committed:
Setup cypress and add tests for homepage

dan-niles committed:
Update CONTRIBUTING.md with Cypress instructions

dan-niles committed:
Rename test job for zimui in Tests.yml

dan-niles committed:
Remove default channel banner in scraper

benoit74 committed:
Merge pull request #225 from openzim/ui-revamp-homepage

Create homepage UI for channels in `zimui`

dan-niles committed:
Update Author schema to include channel description and joined date

dan-niles committed:
Setup video.js and add video player page

dan-niles committed:
Remove locale argument from entrypoint.py

benoit74 committed:
Merge pull request #232 from openzim/remove-locale-arg

Finish removal of locale related code

benoit74 committed:
Merge pull request #229 from openzim/video-player-page

Add Video player page

dependabot[bot] committed:
Bump ws from 8.17.0 to 8.17.1 in /zimui

Bumps [ws](https://github.com/websockets/ws) from 8.17.0 to 8.17.1.
- [Release notes](https://github.com/websockets/ws/releases)
- [Commits](https://github.com/websockets/ws/compare/8.17.0...8.17.1)

---
updated-dependencies:
- dependency-name: ws
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

benoit74 committed:
Merge pull request #234 from openzim/dependabot/npm_and_yarn/zimui/ws-8.17.1

Bump ws from 8.17.0 to 8.17.1 in /zimui

dan-niles committed:
Update error handling for unknown videos, playlists and paths

dan-niles committed:
Upgrade dependencies, including scraperlib 3.4.0

dan-niles committed:
Implement bootstrap v1.0.2

benoit74 committed:
Ignore codecov status until real unit testing is implemented is scraper

benoit74 committed:
Merge pull request #239 from openzim/scraperlib_3_4_0_v3

Upgrade to scraperlib v3.4.0 to use VP9

dan-niles committed:
Add playlist panel to video player page

benoit74 committed:
Merge pull request #237 from openzim/playlist-player

Add playlist panel to video player page

dan-niles committed:
Remove CLI argument for autoplay

dan-niles committed:
Remove unused files from templates/assets folder

benoit74 committed:
Merge pull request #244 from openzim/remove-autoplay-arg

Remove `autoplay` CLI argument

dan-niles committed:
Mock video fetching for Cypress tests

dan-niles committed:
Add individual playlist view page

benoit74 committed:
Merge pull request #246 from openzim/fix-cypress-test

Mock video fetching for Cypress tests

benoit74 committed:
Merge pull request #247 from openzim/playlist-view-page

Add individual playlist view page

dan-niles committed:
Fix new line rendering in descriptions in zimui (#248)

benoit74 committed:
Merge pull request #249 from openzim/fix-newline

Fix new line rendering in descriptions in zimui

dan-niles committed:
Fix ZIM title in index.html of new Vue.js UI

dan-niles committed:
Display playlist panel for videos played from homepage

dan-niles committed:
Fix side margins on tablet screens in zimui

dan-niles committed:
Calculate playlist panel container height dynamically

benoit74 committed:
Merge pull request #255 from openzim/fix-tablet-margin

Fix side margins on tablet screens in zimui

benoit74 committed:
Merge pull request #251 from openzim/homepage-playlist-panel

Display playlist panel for videos played from homepage

benoit74 committed:
Fix codecov configuration to not fail CI if coverage is not OK for now

benoit74 committed:
Merge pull request #250 from openzim/fix-zim-title

Fix title in index.html to be ZIM title

benoit74 committed:
Fix codecov configuration file

dan-niles committed:
Setup integration test and workflow

dan-niles committed:
Add "type" field to package.json and rename cypress config

dan-niles committed:
Update dependencies and add support for ogv.js in video player

benoit74 committed:
Merge pull request #258 from openzim/add-integration-test

Setup integration tests and workflow

dan-niles committed:
Bump Github actions

benoit74 committed:
Merge pull request #260 from openzim/bump-github-actions

Bump Github actions

benoit74 committed:
Merge pull request #259 from openzim/setup-ogvjs

Add support for ogv.js in video player

dan-niles committed:
Remove CLI argument for `no-zim`

dan-niles committed:
Validate if ZIM file is creatable at the output directory

dan-niles committed:
Add downloaded videos and thumbnails to ZIM on the fly

dan-niles committed:
Add channel branding to ZIM on the fly

dan-niles committed:
Add video subtitles to ZIM on the fly

dan-niles committed:
Remove 'add_files_to_zim' method from Youtube2Zim class

dan-niles committed:
Add debug logs in 'add_file_to_zim' and update CHANGELOG

dan-niles committed:
Refactor scraper.py and optimize ZIM file creation process

dan-niles committed:
Remove '--keep' CLI argument

benoit74 committed:
Merge pull request #262 from openzim/add-to-zim-on-the-fly

Add items to the ZIM "on the fly"

dan-niles committed:
Save main and secondary color in config.json

dan-niles committed:
Update Vuetify initialization to load primary and secondary colors from config.json

dan-niles committed:
Update UI to support different background colors from config.json

dan-niles committed:
Update Config class in schemas.py

benoit74 committed:
Merge pull request #263 from openzim/configure-ui-colors

Add support for color CLI arguments in new UI

dan-niles committed:
Add playlist count to Channel schema

dan-niles committed:
Hide tabs if there is only one playlist in zimui

benoit74 committed:
Merge pull request #266 from openzim/hide-playlists-tab

Hide playlists tab in `zimui` if there is only 1 playlist

dan-niles committed:
Index playlists and videos metadata using custom IndexData

benoit74 committed:
Merge pull request #265 from openzim/restore-search-functionality

Restore search functionality by indexing content in JSON files

dan-niles committed:
Fix channel title truncation for smaller screens in zimui

benoit74 committed:
Merge pull request #267 from openzim/fix-channel-title-truncation

Fix channel title truncation for smaller screens in zimui

dan-niles committed:
Add e2e tests for zim metadata, channel.json, videos, and playlists

dan-niles committed:
Add subtitle file validation in integration tests

dan-niles committed:
Update CHANGELOG mentioning addition of integration tests

benoit74 committed:
Merge pull request #268 from openzim/update-integration-tests

Update integration tests to check content in ZIM

dan-niles committed:
Update README (#269)

dan-niles committed:
Update Vite configuration in zimui for native ES6 support

dan-niles committed:
Update README

benoit74 committed:
Merge pull request #271 from openzim/update-readme

Update README (#269)

dan-niles committed:
Tweak UI to work with older browsers

dan-niles committed:
Add resize-observer-polyfill for older browsers

dan-niles committed:
Polyfill thumbnails for browsers without WebP support

dan-niles committed:
Update video player component with fallback to native browser video playback

dan-niles committed:
Downgrade ogv dependency from version 1.9.0 to 1.8.9

dan-niles committed:
Update zimui index.html with JavaScript disabled message

benoit74 committed:
Merge pull request #272 from openzim/add-es6-support

Update Vite configuration in zimui for native ES6 support

dan-niles committed:
Add an overlay image for the channel banner

benoit74 committed:
Merge pull request #280 from openzim/add-banner-overlay

Add an overlay image for the channel banner

benoit74 committed:
Release 3.0.0

benoit74 committed:
Prepare for 3.0.1

benoit74 committed:
Fix #281: Remove README disclaimer now that 3.0 is stable

dan-niles committed:
Disable preloading of subtitles in video.js

benoit74 committed:
Merge pull request #283 from openzim/disable-subtitles-preloading

Disable preloading of subtitles in video.js

dan-niles committed:
Refactor scraper.py to exit properly when exceptions are raised

dan-niles committed:
Implement infinite scroll for video/playlist lists

dan-niles committed:
Move all code inside the run method into a try block

dan-niles committed:
Update CHANGELOG

benoit74 committed:
Merge pull request #288 from openzim/fix-exception-issue

Refactor scraper to exit properly when exceptions are raised

dan-niles committed:
Update CHANGELOG

dan-niles committed:
Change isLoading to ref and use length to find end of list during infinite scroll

benoit74 committed:
Merge pull request #289 from openzim/implement-infinite-scroll

Implement infinite scroll for video/playlist lists

dan-niles committed:
Bump video.js to v8.17.3

benoit74 committed:
Merge pull request #290 from openzim/bump-videojs

Bump video.js to v8.17.3

benoit74 committed:
Release 3.0.1

benoit74 committed:
Prepare for 3.0.2

dan-niles committed:
Report scraper progress in a JSON file

dan-niles committed:
Merge pull request #294 from openzim/scraper-progress

Report scraper progress in a JSON file

dependabot[bot] committed:
Bump axios from 1.7.2 to 1.7.4 in /zimui

Bumps [axios](https://github.com/axios/axios) from 1.7.2 to 1.7.4.
- [Release notes](https://github.com/axios/axios/releases)
- [Changelog](https://github.com/axios/axios/blob/v1.x/CHANGELOG.md)
- [Commits](https://github.com/axios/axios/compare/v1.7.2...v1.7.4)

---
updated-dependencies:
- dependency-name: axios
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dan-niles committed:
Refactor playlist extraction logic to maintain order

dan-niles committed:
Merge pull request #296 from openzim/fix-main-playlist

Fix main playlist selection to respect the order of provided playlist IDs

dan-niles committed:
Update `zimui` title dynamically with the selected playlist/video title

dan-niles committed:
Fix `PLAYLISTS` tab not being highlighted when the page is reloaded

benoit74 committed:
Merge pull request #300 from openzim/fix-zim-titles

Update zimui HTML page title dynamically with playlist/video title

benoit74 committed:
Merge pull request #295 from openzim/dependabot/npm_and_yarn/zimui/axios-1.7.4

Bump axios from 1.7.2 to 1.7.4 in /zimui

benoit74 committed:
Create dependabot.yml

Configure dependabot to run daily and group dependencies updates

dependabot[bot] committed:
Bump the production-dependencies group in /zimui with 8 updates

Bumps the production-dependencies group in /zimui with 8 updates:

| Package | From | To |
| --- | --- | --- |
| [@fontsource/roboto](https://github.com/fontsource/font-files/tree/HEAD/fonts/google/roboto) | `5.0.13` | `5.0.14` |
| [axios](https://github.com/axios/axios) | `1.7.4` | `1.7.7` |
| [dayjs](https://github.com/iamkun/dayjs) | `1.11.11` | `1.11.13` |
| [pinia](https://github.com/vuejs/pinia) | `2.1.7` | `2.2.2` |
| [vite-plugin-vuetify](https://github.com/vuetifyjs/vuetify-loader) | `2.0.3` | `2.0.4` |
| [vue](https://github.com/vuejs/core) | `3.4.27` | `3.4.38` |
| [vue-router](https://github.com/vuejs/router) | `4.3.2` | `4.4.3` |
| [vuetify](https://github.com/vuetifyjs/vuetify/tree/HEAD/packages/vuetify) | `3.6.7` | `3.7.1` |


Updates `@fontsource/roboto` from 5.0.13 to 5.0.14
- [Changelog](https://github.com/fontsource/font-files/blob/main/fonts/google/roboto/CHANGELOG.md)
- [Commits](https://github.com/fontsource/font-files/commits/HEAD/fonts/google/roboto)

Updates `axios` from 1.7.4 to 1.7.7
- [Release notes](https://github.com/axios/axios/releases)
- [Changelog](https://github.com/axios/axios/blob/v1.x/CHANGELOG.md)
- [Commits](https://github.com/axios/axios/compare/v1.7.4...v1.7.7)

Updates `dayjs` from 1.11.11 to 1.11.13
- [Release notes](https://github.com/iamkun/dayjs/releases)
- [Changelog](https://github.com/iamkun/dayjs/blob/v1.11.13/CHANGELOG.md)
- [Commits](https://github.com/iamkun/dayjs/compare/v1.11.11...v1.11.13)

Updates `pinia` from 2.1.7 to 2.2.2
- [Release notes](https://github.com/vuejs/pinia/releases)
- [Commits](https://github.com/vuejs/pinia/compare/pinia@2.1.7...pinia@2.2.2)

Updates `vite-plugin-vuetify` from 2.0.3 to 2.0.4
- [Release notes](https://github.com/vuetifyjs/vuetify-loader/releases)
- [Commits](https://github.com/vuetifyjs/vuetify-loader/compare/vite-plugin-vuetify@2.0.3...vite-plugin-vuetify@2.0.4)

Updates `vue` from 3.4.27 to 3.4.38
- [Release notes](https://github.com/vuejs/core/releases)
- [Changelog](https://github.com/vuejs/core/blob/v3.4.38/CHANGELOG.md)
- [Commits](https://github.com/vuejs/core/compare/v3.4.27...v3.4.38)

Updates `vue-router` from 4.3.2 to 4.4.3
- [Release notes](https://github.com/vuejs/router/releases)
- [Commits](https://github.com/vuejs/router/compare/v4.3.2...v4.4.3)

Updates `vuetify` from 3.6.7 to 3.7.1
- [Release notes](https://github.com/vuetifyjs/vuetify/releases)
- [Commits](https://github.com/vuetifyjs/vuetify/commits/v3.7.1/packages/vuetify)

---
updated-dependencies:
- dependency-name: "@fontsource/roboto"
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: production-dependencies
- dependency-name: axios
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: production-dependencies
- dependency-name: dayjs
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: production-dependencies
- dependency-name: pinia
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: production-dependencies
- dependency-name: vite-plugin-vuetify
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: production-dependencies
- dependency-name: vue
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: production-dependencies
- dependency-name: vue-router
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: production-dependencies
- dependency-name: vuetify
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: production-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump the production-dependencies group in /scraper with 9 updates

Bumps the production-dependencies group in /scraper with 9 updates:

| Package | From | To |
| --- | --- | --- |
| [zimscraperlib](https://github.com/openzim/python-scraperlib) | `3.4.0` | `4.0.0` |
| [pydantic](https://github.com/pydantic/pydantic) | `2.7.2` | `2.8.2` |
| [black](https://github.com/psf/black) | `24.4.2` | `24.8.0` |
| [ruff](https://github.com/astral-sh/ruff) | `0.4.10` | `0.6.3` |
| [pyright](https://github.com/RobertCraigie/pyright-python) | `1.1.371` | `1.1.378` |
| [pytest](https://github.com/pytest-dev/pytest) | `8.2.2` | `8.3.2` |
| [coverage](https://github.com/nedbat/coveragepy) | `7.5.3` | `7.6.1` |
| [pre-commit](https://github.com/pre-commit/pre-commit) | `3.7.1` | `3.8.0` |
| [debugpy](https://github.com/microsoft/debugpy) | `1.8.1` | `1.8.5` |


Updates `zimscraperlib` from 3.4.0 to 4.0.0
- [Release notes](https://github.com/openzim/python-scraperlib/releases)
- [Changelog](https://github.com/openzim/python-scraperlib/blob/main/CHANGELOG.md)
- [Commits](https://github.com/openzim/python-scraperlib/compare/v3.4.0...v4.0.0)

Updates `pydantic` from 2.7.2 to 2.8.2
- [Release notes](https://github.com/pydantic/pydantic/releases)
- [Changelog](https://github.com/pydantic/pydantic/blob/main/HISTORY.md)
- [Commits](https://github.com/pydantic/pydantic/compare/v2.7.2...v2.8.2)

Updates `black` from 24.4.2 to 24.8.0
- [Release notes](https://github.com/psf/black/releases)
- [Changelog](https://github.com/psf/black/blob/main/CHANGES.md)
- [Commits](https://github.com/psf/black/compare/24.4.2...24.8.0)

Updates `ruff` from 0.4.10 to 0.6.3
- [Release notes](https://github.com/astral-sh/ruff/releases)
- [Changelog](https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md)
- [Commits](https://github.com/astral-sh/ruff/compare/v0.4.10...0.6.3)

Updates `pyright` from 1.1.371 to 1.1.378
- [Release notes](https://github.com/RobertCraigie/pyright-python/releases)
- [Commits](https://github.com/RobertCraigie/pyright-python/compare/v1.1.371...v1.1.378)

Updates `pytest` from 8.2.2 to 8.3.2
- [Release notes](https://github.com/pytest-dev/pytest/releases)
- [Changelog](https://github.com/pytest-dev/pytest/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pytest-dev/pytest/compare/8.2.2...8.3.2)

Updates `coverage` from 7.5.3 to 7.6.1
- [Release notes](https://github.com/nedbat/coveragepy/releases)
- [Changelog](https://github.com/nedbat/coveragepy/blob/master/CHANGES.rst)
- [Commits](https://github.com/nedbat/coveragepy/compare/7.5.3...7.6.1)

Updates `pre-commit` from 3.7.1 to 3.8.0
- [Release notes](https://github.com/pre-commit/pre-commit/releases)
- [Changelog](https://github.com/pre-commit/pre-commit/blob/main/CHANGELOG.md)
- [Commits](https://github.com/pre-commit/pre-commit/compare/v3.7.1...v3.8.0)

Updates `debugpy` from 1.8.1 to 1.8.5
- [Release notes](https://github.com/microsoft/debugpy/releases)
- [Commits](https://github.com/microsoft/debugpy/compare/v1.8.1...v1.8.5)

---
updated-dependencies:
- dependency-name: zimscraperlib
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: production-dependencies
- dependency-name: pydantic
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: production-dependencies
- dependency-name: black
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: production-dependencies
- dependency-name: ruff
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: production-dependencies
- dependency-name: pyright
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: production-dependencies
- dependency-name: pytest
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: production-dependencies
- dependency-name: coverage
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: production-dependencies
- dependency-name: pre-commit
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: production-dependencies
- dependency-name: debugpy
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: production-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump the development-dependencies group in /zimui with 17 updates

Bumps the development-dependencies group in /zimui with 17 updates:

| Package | From | To |
| --- | --- | --- |
| [@rushstack/eslint-patch](https://github.com/microsoft/rushstack/tree/HEAD/eslint/eslint-patch) | `1.10.3` | `1.10.4` |
| [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) | `20.12.12` | `22.5.2` |
| [@vitejs/plugin-legacy](https://github.com/vitejs/vite/tree/HEAD/packages/plugin-legacy) | `5.4.1` | `5.4.2` |
| [@vitejs/plugin-vue](https://github.com/vitejs/vite-plugin-vue/tree/HEAD/packages/plugin-vue) | `5.0.4` | `5.1.3` |
| [cypress](https://github.com/cypress-io/cypress) | `13.11.0` | `13.14.1` |
| [eslint](https://github.com/eslint/eslint) | `8.57.0` | `9.9.1` |
| [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) | `9.26.0` | `9.28.0` |
| [jsdom](https://github.com/jsdom/jsdom) | `24.1.0` | `25.0.0` |
| [npm-run-all2](https://github.com/bcomnes/npm-run-all2) | `6.2.0` | `6.2.2` |
| [ogv](https://github.com/bvibber/ogv.js) | `1.8.9` | `1.9.0` |
| [prettier](https://github.com/prettier/prettier) | `3.2.5` | `3.3.3` |
| [terser](https://github.com/terser/terser) | `5.31.3` | `5.31.6` |
| [typescript](https://github.com/Microsoft/TypeScript) | `5.4.5` | `5.5.4` |
| [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) | `5.2.11` | `5.4.2` |
| [vite-plugin-static-copy](https://github.com/sapphi-red/vite-plugin-static-copy) | `1.0.5` | `1.0.6` |
| [vitest](https://github.com/vitest-dev/vitest/tree/HEAD/packages/vitest) | `1.6.0` | `2.0.5` |
| [vue-tsc](https://github.com/vuejs/language-tools/tree/HEAD/packages/tsc) | `2.0.19` | `2.1.4` |

Updates `@rushstack/eslint-patch` from 1.10.3 to 1.10.4
- [Changelog](https://github.com/microsoft/rushstack/blob/main/eslint/eslint-patch/CHANGELOG.md)
- [Commits](https://github.com/microsoft/rushstack/commits/@rushstack/eslint-patch_v1.10.4/eslint/eslint-patch)

Updates `@types/node` from 20.12.12 to 22.5.2
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

Updates `@vitejs/plugin-legacy` from 5.4.1 to 5.4.2
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/main/packages/plugin-legacy/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v5.4.2/packages/plugin-legacy)

Updates `@vitejs/plugin-vue` from 5.0.4 to 5.1.3
- [Release notes](https://github.com/vitejs/vite-plugin-vue/releases)
- [Changelog](https://github.com/vitejs/vite-plugin-vue/blob/main/packages/plugin-vue/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite-plugin-vue/commits/plugin-vue@5.1.3/packages/plugin-vue)

Updates `cypress` from 13.11.0 to 13.14.1
- [Release notes](https://github.com/cypress-io/cypress/releases)
- [Changelog](https://github.com/cypress-io/cypress/blob/develop/CHANGELOG.md)
- [Commits](https://github.com/cypress-io/cypress/compare/v13.11.0...v13.14.1)

Updates `eslint` from 8.57.0 to 9.9.1
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.57.0...v9.9.1)

Updates `eslint-plugin-vue` from 9.26.0 to 9.28.0
- [Release notes](https://github.com/vuejs/eslint-plugin-vue/releases)
- [Commits](https://github.com/vuejs/eslint-plugin-vue/compare/v9.26.0...v9.28.0)

Updates `jsdom` from 24.1.0 to 25.0.0
- [Release notes](https://github.com/jsdom/jsdom/releases)
- [Changelog](https://github.com/jsdom/jsdom/blob/main/Changelog.md)
- [Commits](https://github.com/jsdom/jsdom/compare/24.1.0...25.0.0)

Updates `npm-run-all2` from 6.2.0 to 6.2.2
- [Release notes](https://github.com/bcomnes/npm-run-all2/releases)
- [Changelog](https://github.com/bcomnes/npm-run-all2/blob/master/CHANGELOG.md)
- [Commits](https://github.com/bcomnes/npm-run-all2/compare/v6.2.0...v6.2.2)

Updates `ogv` from 1.8.9 to 1.9.0
- [Release notes](https://github.com/bvibber/ogv.js/releases)
- [Changelog](https://github.com/bvibber/ogv.js/blob/main/CHANGES.md)
- [Commits](https://github.com/bvibber/ogv.js/compare/1.8.9...1.9.0)

Updates `prettier` from 3.2.5 to 3.3.3
- [Release notes](https://github.com/prettier/prettier/releases)
- [Changelog](https://github.com/prettier/prettier/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prettier/prettier/compare/3.2.5...3.3.3)

Updates `terser` from 5.31.3 to 5.31.6
- [Changelog](https://github.com/terser/terser/blob/master/CHANGELOG.md)
- [Commits](https://github.com/terser/terser/commits)

Updates `typescript` from 5.4.5 to 5.5.4
- [Release notes](https://github.com/Microsoft/TypeScript/releases)
- [Changelog](https://github.com/microsoft/TypeScript/blob/main/azure-pipelines.release.yml)
- [Commits](https://github.com/Microsoft/TypeScript/compare/v5.4.5...v5.5.4)

Updates `vite` from 5.2.11 to 5.4.2
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/main/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v5.4.2/packages/vite)

Updates `vite-plugin-static-copy` from 1.0.5 to 1.0.6
- [Release notes](https://github.com/sapphi-red/vite-plugin-static-copy/releases)
- [Changelog](https://github.com/sapphi-red/vite-plugin-static-copy/blob/main/CHANGELOG.md)
- [Commits](https://github.com/sapphi-red/vite-plugin-static-copy/compare/vite-plugin-static-copy@1.0.5...vite-plugin-static-copy@1.0.6)

Updates `vitest` from 1.6.0 to 2.0.5
- [Release notes](https://github.com/vitest-dev/vitest/releases)
- [Commits](https://github.com/vitest-dev/vitest/commits/v2.0.5/packages/vitest)

Updates `vue-tsc` from 2.0.19 to 2.1.4
- [Release notes](https://github.com/vuejs/language-tools/releases)
- [Changelog](https://github.com/vuejs/language-tools/blob/master/CHANGELOG.md)
- [Commits](https://github.com/vuejs/language-tools/commits/v2.1.4/packages/tsc)

---
updated-dependencies:
- dependency-name: "@rushstack/eslint-patch"
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: development-dependencies
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: development-dependencies
- dependency-name: "@vitejs/plugin-legacy"
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: development-dependencies
- dependency-name: "@vitejs/plugin-vue"
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: development-dependencies
- dependency-name: cypress
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: development-dependencies
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: development-dependencies
- dependency-name: eslint-plugin-vue
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: development-dependencies
- dependency-name: jsdom
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: development-dependencies
- dependency-name: npm-run-all2
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: development-dependencies
- dependency-name: ogv
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: development-dependencies
- dependency-name: prettier
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: development-dependencies
- dependency-name: terser
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: development-dependencies
- dependency-name: typescript
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: development-dependencies
- dependency-name: vite
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: development-dependencies
- dependency-name: vite-plugin-static-copy
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: development-dependencies
- dependency-name: vitest
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: development-dependencies
- dependency-name: vue-tsc
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: development-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

benoit74 committed:
Adapt to zimscraperlib 4

benoit74 committed:
Rollback upgrade to eslint 9, failed to make it work properly with Vue.JS and typescript

benoit74 committed:
Update CHANGELOG

benoit74 committed:
Merge pull request #306 from openzim/upgrade_deps

Upgrade dependencies, including zimscraperlib 4

dependabot[bot] committed:
Bump pyright

Bumps the production-dependencies group with 1 update in the /scraper directory: [pyright](https://github.com/RobertCraigie/pyright-python).


Updates `pyright` from 1.1.378 to 1.1.379
- [Release notes](https://github.com/RobertCraigie/pyright-python/releases)
- [Commits](https://github.com/RobertCraigie/pyright-python/compare/v1.1.378...v1.1.379)

---
updated-dependencies:
- dependency-name: pyright
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: production-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump vue

Bumps the production-dependencies group with 1 update in the /zimui directory: [vue](https://github.com/vuejs/core).


Updates `vue` from 3.4.38 to 3.5.1
- [Release notes](https://github.com/vuejs/core/releases)
- [Changelog](https://github.com/vuejs/core/blob/main/CHANGELOG.md)
- [Commits](https://github.com/vuejs/core/compare/v3.4.38...v3.5.1)

---
updated-dependencies:
- dependency-name: vue
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: production-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump the development-dependencies group across 1 directory with 5 updates

Bumps the development-dependencies group with 5 updates in the /zimui directory:

| Package | From | To |
| --- | --- | --- |
| [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) | `22.5.2` | `22.5.4` |
| [cypress](https://github.com/cypress-io/cypress) | `13.14.1` | `13.14.2` |
| [eslint](https://github.com/eslint/eslint) | `8.57.0` | `9.9.1` |
| [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) | `5.4.2` | `5.4.3` |
| [vue-tsc](https://github.com/vuejs/language-tools/tree/HEAD/packages/tsc) | `2.1.4` | `2.1.6` |

Updates `@types/node` from 22.5.2 to 22.5.4
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

Updates `cypress` from 13.14.1 to 13.14.2
- [Release notes](https://github.com/cypress-io/cypress/releases)
- [Changelog](https://github.com/cypress-io/cypress/blob/develop/CHANGELOG.md)
- [Commits](https://github.com/cypress-io/cypress/compare/v13.14.1...v13.14.2)

Updates `eslint` from 8.57.0 to 9.9.1
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.57.0...v9.9.1)

Updates `vite` from 5.4.2 to 5.4.3
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/main/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v5.4.3/packages/vite)

Updates `vue-tsc` from 2.1.4 to 2.1.6
- [Release notes](https://github.com/vuejs/language-tools/releases)
- [Changelog](https://github.com/vuejs/language-tools/blob/master/CHANGELOG.md)
- [Commits](https://github.com/vuejs/language-tools/commits/v2.1.6/packages/tsc)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: development-dependencies
- dependency-name: cypress
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: development-dependencies
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: development-dependencies
- dependency-name: vite
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: development-dependencies
- dependency-name: vue-tsc
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: development-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

benoit74 committed:
Rollback upgrade to eslint 9, failed to make it work properly with Vue.JS and typescript

benoit74 committed:
Merge pull request #314 from openzim/upgrade-deps

Upgrade more deps

benoit74 committed:
Ignore eslint 9.x and next for now

benoit74 committed:
Release 3.1.0

benoit74 committed:
Prepare for 3.1.1

dependabot[bot] committed:
Bump vue

Bumps the production-dependencies group with 1 update in the /zimui directory: [vue](https://github.com/vuejs/core).


Updates `vue` from 3.5.1 to 3.5.3
- [Release notes](https://github.com/vuejs/core/releases)
- [Changelog](https://github.com/vuejs/core/blob/main/CHANGELOG.md)
- [Commits](https://github.com/vuejs/core/compare/v3.5.1...v3.5.3)

---
updated-dependencies:
- dependency-name: vue
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: production-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump the production-dependencies group across 1 directory with 2 updates

Bumps the production-dependencies group with 2 updates in the /scraper directory: [pydantic](https://github.com/pydantic/pydantic) and [ruff](https://github.com/astral-sh/ruff).


Updates `pydantic` from 2.8.2 to 2.9.0
- [Release notes](https://github.com/pydantic/pydantic/releases)
- [Changelog](https://github.com/pydantic/pydantic/blob/main/HISTORY.md)
- [Commits](https://github.com/pydantic/pydantic/compare/v2.8.2...v2.9.0)

Updates `ruff` from 0.6.3 to 0.6.4
- [Release notes](https://github.com/astral-sh/ruff/releases)
- [Changelog](https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md)
- [Commits](https://github.com/astral-sh/ruff/compare/0.6.3...0.6.4)

---
updated-dependencies:
- dependency-name: pydantic
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: production-dependencies
- dependency-name: ruff
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: production-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

benoit74 committed:
Merge pull request #322 from openzim/update

Update dependencies

dependabot[bot] committed:
Bump the production-dependencies group across 1 directory with 3 updates

Bumps the production-dependencies group with 3 updates in the /scraper directory: [pydantic](https://github.com/pydantic/pydantic), [pyright](https://github.com/RobertCraigie/pyright-python) and [pytest](https://github.com/pytest-dev/pytest).


Updates `pydantic` from 2.9.0 to 2.9.1
- [Release notes](https://github.com/pydantic/pydantic/releases)
- [Changelog](https://github.com/pydantic/pydantic/blob/main/HISTORY.md)
- [Commits](https://github.com/pydantic/pydantic/compare/v2.9.0...v2.9.1)

Updates `pyright` from 1.1.379 to 1.1.380
- [Release notes](https://github.com/RobertCraigie/pyright-python/releases)
- [Commits](https://github.com/RobertCraigie/pyright-python/compare/v1.1.379...v1.1.380)

Updates `pytest` from 8.3.2 to 8.3.3
- [Release notes](https://github.com/pytest-dev/pytest/releases)
- [Changelog](https://github.com/pytest-dev/pytest/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pytest-dev/pytest/compare/8.3.2...8.3.3)

---
updated-dependencies:
- dependency-name: pydantic
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: production-dependencies
- dependency-name: pyright
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: production-dependencies
- dependency-name: pytest
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: production-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump the production-dependencies group across 1 directory with 4 updates

Bumps the production-dependencies group with 4 updates in the /zimui directory: [@fontsource/roboto](https://github.com/fontsource/font-files/tree/HEAD/fonts/google/roboto), [video.js](https://github.com/videojs/video.js), [vue](https://github.com/vuejs/core) and [vue-router](https://github.com/vuejs/router).


Updates `@fontsource/roboto` from 5.0.14 to 5.1.0
- [Changelog](https://github.com/fontsource/font-files/blob/main/fonts/google/roboto/CHANGELOG.md)
- [Commits](https://github.com/fontsource/font-files/commits/HEAD/fonts/google/roboto)

Updates `video.js` from 8.17.3 to 8.17.4
- [Release notes](https://github.com/videojs/video.js/releases)
- [Changelog](https://github.com/videojs/video.js/blob/main/CHANGELOG.md)
- [Commits](https://github.com/videojs/video.js/compare/v8.17.3...v8.17.4)

Updates `vue` from 3.5.3 to 3.5.4
- [Release notes](https://github.com/vuejs/core/releases)
- [Changelog](https://github.com/vuejs/core/blob/main/CHANGELOG.md)
- [Commits](https://github.com/vuejs/core/compare/v3.5.3...v3.5.4)

Updates `vue-router` from 4.4.3 to 4.4.4
- [Release notes](https://github.com/vuejs/router/releases)
- [Commits](https://github.com/vuejs/router/compare/v4.4.3...v4.4.4)

---
updated-dependencies:
- dependency-name: "@fontsource/roboto"
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: production-dependencies
- dependency-name: video.js
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: production-dependencies
- dependency-name: vue
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: production-dependencies
- dependency-name: vue-router
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: production-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump the development-dependencies group across 1 directory with 4 updates

Bumps the development-dependencies group with 4 updates in the /zimui directory: [terser](https://github.com/terser/terser), [typescript](https://github.com/microsoft/TypeScript), [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) and [vitest](https://github.com/vitest-dev/vitest/tree/HEAD/packages/vitest).


Updates `terser` from 5.31.6 to 5.32.0
- [Changelog](https://github.com/terser/terser/blob/master/CHANGELOG.md)
- [Commits](https://github.com/terser/terser/compare/v5.31.6...v5.32.0)

Updates `typescript` from 5.5.4 to 5.6.2
- [Release notes](https://github.com/microsoft/TypeScript/releases)
- [Changelog](https://github.com/microsoft/TypeScript/blob/main/azure-pipelines.release.yml)
- [Commits](https://github.com/microsoft/TypeScript/compare/v5.5.4...v5.6.2)

Updates `vite` from 5.4.3 to 5.4.4
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/v5.4.4/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v5.4.4/packages/vite)

Updates `vitest` from 2.0.5 to 2.1.0
- [Release notes](https://github.com/vitest-dev/vitest/releases)
- [Commits](https://github.com/vitest-dev/vitest/commits/v2.1.0/packages/vitest)

---
updated-dependencies:
- dependency-name: terser
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: development-dependencies
- dependency-name: typescript
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: development-dependencies
- dependency-name: vite
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: development-dependencies
- dependency-name: vitest
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: development-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

benoit74 committed:
Merge pull request #334 from openzim/update

Update dependencies

benoit74 committed:
Delete .github/dependabot.yml

Remove dependabot, this is creating too much noise / risk for the scraper stability

benoit74 committed:
Merge behaviors of user/channel types

benoit74 committed:
Rename CHANGELOG to CHANGELOG.md

benoit74 committed:
Ignore empty playlists

benoit74 committed:
Merge pull request #341 from openzim/empty_playlist

Ignore empty playlists

benoit74 committed:
Merge pull request #339 from openzim/merge_user_channel

Merge behaviors of user/channel types and add support for `forHandle`

dan-niles committed:
Update scraper to generate home_playlists.json

dan-niles committed:
Update videos tab to display all playlists in carousels

dan-niles committed:
Update cypress tests for new videos tab layout

dan-niles committed:
Update CHANGELOG.md

benoit74 committed:
Enhance yt_dlp format and fix bug

benoit74 committed:
Merge pull request #355 from openzim/change_yt_dlp_format

Enhance yt_dlp format and fix bug

benoit74 committed:
Update CHANGELOG with missing entry from last PR #355

benoit74 committed:
Merge pull request #349 from openzim/new-landing-page-ui

Update layout of `Videos` tab in `zimui` to display videos from all playlists in the ZIM

benoit74 committed:
Update README: adapt to recent changes and move details to Wiki

benoit74 committed:
Release 3.2.0

benoit74 committed:
Prepare for 3.2.1

benoit74 committed:
Filter-out non-public videos and properly cleanup unsuccessful videos

dan-niles committed:
Raise exception if there are no videos in the playlists

benoit74 committed:
Merge pull request #365 from openzim/handle-empty-playlists

Raise exception if there are no videos in the playlists

benoit74 committed:
Merge pull request #363 from openzim/filter_private_vids

Filter-out non-public videos and properly cleanup unsuccessful videos

benoit74 committed:
Use proper ZIM metadata key for `Scraper` and `Tags`

benoit74 committed:
Merge pull request #370 from openzim/fix_tags

Use proper ZIM metadata key for `Scraper` and `Tags`

arjitcodes committed:
Shorts, lives and long videos in the UI + fix shorts display

benoit74 committed:
Simplify scraper by dropping --type CLI argument

benoit74 committed:
Always reencode using our presets (even for high quality) and choose best format when downloading from Youtube

benoit74 committed:
Add missing playsinline attribute for proper operation on iOS

benoit74 committed:
Merge pull request #372 from openzim/drop_type_arg

Simplify scraper by dropping `--type` CLI argument

benoit74 committed:
Merge pull request #373 from openzim/format_simplified

Always reencode using our presets (even for high quality) and choose best format

benoit74 committed:
Merge pull request #374 from openzim/playsinline

Add missing playsinline attribute for proper operation on iOS

benoit74 committed:
Release 3.2.1

benoit74 committed:
Prepare for 3.2.2

benoit74 committed:
Check for empty playlists after filtering, and after downloading videos

benoit74 committed:
Merge pull request #376 from openzim/empty_playlist

Check for empty playlists after filtering, and after downloading videos

benoit74 committed:
Fix Docker badge

benoit74 committed:
Fix / enrich integration tests

benoit74 committed:
Merge pull request #378 from openzim/fix/differentiate-shorts-lives-normal-videos

Shorts, lives and long videos in the UI + fix shorts display

benoit74 committed:
Unlisted videos must also be included inside the ZIM

benoit74 committed:
Merge pull request #379 from openzim/add_unlisted_videos

Unlisted videos must also be included inside the ZIM

benoit74 committed:
Release 3.3.0

benoit74 committed:
Prepare for 3.3.1

arjitcodes committed:
feat: Add support for video chapters

benoit74 committed:
Merge pull request #388 from arjitcodes/feature/support-video-chapters

Feat: support video chapters

abdulazimRabie committed:
feat:Activate links in video description

benoit74 committed:
Merge pull request #391 from abdulazimRabie/feature/clickable-desc-links

feat:Activate links in video description

arjitcodes committed:
feat: add skip controls & keyboard/touch shortcuts

benoit74 committed:
Merge pull request #398 from arjitcodes/enhance/player-interactions

Add skip controls & keyboard/touch shortcuts

abdulazimRabie committed:
enhance: add theater mode to video player

benoit74 committed:
Merge pull request #400 from abdulazimRabie/enhance/theater-mode

Add theater mode to video player

benoit74 committed:
Fix detection of credentials validity

benoit74 committed:
Merge pull request #403 from openzim/fix_credentials_check

Fix detection of credentials validity

benoit74 committed:
Upgrade Python (3.13, zimscraperlib 5.1.1)

benoit74 committed:
Merge pull request #404 from openzim/upgrade-python

Upgrade Python (3.13, zimscraperlib 5.1.1)

benoit74 committed:
Release 3.4.0

benoit74 committed:
Upgrade CI

benoit74 committed:
Merge pull request #406 from openzim/upgrade_ci

Upgrade CI

benoit74 committed:
Prepare for 3.4.1

benoit74 committed:
Fix thumbnail processing - 5.1.1 migration bug + error when not webp

benoit74 committed:
Upgrade frontend dependencies

benoit74 committed:
Merge pull request #409 from openzim/youtube_thumb

Fix thumbnail processing - 5.1.1 migration bug + error when not webp

benoit74 committed:
Merge pull request #411 from openzim/upgrade-frontend

Upgrade frontend dependencies

benoit74 committed:
Release 3.4.1

benoit74 committed:
Prepare for 3.4.2

elfkuzco committed:
set up offliner definitions

elfkuzco committed:
Merge pull request #421 from openzim/offliner-definitions

set up offliner definitions

elfkuzco committed:
add workflow dispatch to offliner-definition ci

salvah22 committed:
skip-reencoding

benoit74 committed:
Merge pull request #415 from salvah22/main

skip reencoding

benoit74 committed:
Merge pull request #422 from openzim/offliner-definition-fixes

add workflow dispatch to offliner-definition ci

benoit74 committed:
Fix wrong `format` field name in offliner-definition

benoit74 committed:
Upgrade to Python 3.14 (and dependencies) and Node.JS 24 (and dependencies)

benoit74 committed:
Merge pull request #426 from openzim/upgrade_deps

Upgrade to Python 3.14 (and dependencies) and Node.JS 24 (and dependencies) and Debian Trixie

benoit74 committed:
Fix addition of deno to the path to be accessible from all shells

benoit74 committed:
Merge pull request #427 from openzim/fix-deno-install

Fix addition of deno to the path to be accessible from all shells

benoit74 committed:
Add impersonation library as suggested by Youtube extracter

benoit74 committed:
Merge pull request #428 from openzim/add-impersonation

Add impersonation library as suggested by Youtube extractor

benoit74 committed:
Release 3.5.0

benoit74 committed:
Prepare for 3.5.1

aaryansinhaa committed:
feat: added total playlist duration

elfkuzco committed:
change profile and banner flags to blob types

elfkuzco committed:
fix error in type for banner flag

benoit74 committed:
Merge pull request #437 from openzim/blob-types

change profile and banner flags to blob types

benoit74 committed:
Merge pull request #435 from aaryansinhaa/main

feat: added total playlist duration

arjitcodes committed:
fix(navigation): resolve home tab underbar not updating

arjitcodes committed:
fix(playlist): resolve Cypress E2E crash and clean up types

arjitcodes committed:
Implement dark mode

benoit74 committed:
Merge pull request #447 from arjitcodes/fix/issue-424-dark-mode

Implement dark mode

benoit74 committed:
Merge pull request #449 from arjitcodes/fix/home-tab-underbar-navigation

fix(navigation): resolve home tab underbar not updating

benoit74 committed:
Merge pull request #451 from arjitcodes/fix/playlist-e2e-crash

Fix: Resolve Cypress E2E crash and prevent premature PlaylistPanel rendering

Pranavjeet-Naidu committed:
feat: add a script for 1-min yt zims size

cking100 committed:
Fix Pattern Error for invalid regex group reference

peniel-09 committed:
fix: raise explicit error when zimui/dist is empty or missing

When zimui/dist has not been built before running the scraper,
the ZIM is produced silently without a homepage. This commit:
- Adds an explicit check in scraper.py that raises a ValueError
  with clear instructions if zimui/dist is missing or empty
- Updates README to document the yarn build prerequisite

Fixes #456

peniel-09 committed:
docs: rework README to push Docker forward and document source setup

benoit74 committed:
Merge pull request #455 from cking100/move-to-python3.14

Fix Pattern Error for invalid regex group reference

cking100 committed:
Bundle Zim UI inside the pip package

lumpy72006 committed:
Also cache subtitles and chapters in S3 with TTL-based expiry

benoit74 committed:
Merge pull request #454 from Pranavjeet-Naidu/main

feat: add a script for 1-min yt zims size

benoit74 committed:
Merge pull request #461 from cking100/include-build-in-sdist-wheel

Bundle Zim UI inside the pip package

benoit74 committed:
Merge pull request #460 from lumpy72006/feat/s3-cache-ttl-expiry

feat:  Also cache subtitles and chapters in S3 with TTL-based expiry

benoit74 committed:
Merge pull request #457 from peniel-09/fix/zimui-dist-empty-check

fix: raise explicit error when zimui/dist is empty or missing

aaryansinhaa committed:
resolves #463

benoit74 committed:
Merge pull request #465 from aaryansinhaa/bugfix/CI-issue

resolves #463: Fixes zimui QA checks

Abhijithmns committed:
feat: add linux/arm64 support to Docker image and CI

benoit74 committed:
Merge pull request #466 from Abhijithmns/multi-arch-support

feat:add linux/arm64 support to Docker image

elfkuzco committed:
rename length constraints to graphemes

elfkuzco committed:
Merge pull request #467 from openzim/length-to-graphemes

rename length constraints to graphemes

elfkuzco committed:
add zim metadata to offliner spec

elfkuzco committed:
Merge pull request #468 from openzim/zim-metadata-offliner-definition

add zim metadata to offliner spec
